sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/datastore/DatastoreV1.java

```diff
@@ -1458,6 +1461,12 @@ public void startBundle(StartBundleContext c) {
     public void processElement(ProcessContext c) throws Exception {
       Mutation write = c.element();
       int size = write.getSerializedSize();
+
+      if (!uniqueMutations.add(c.element())) {
+        batchSize.update(mutations.size());
+        flushBatch();
+      }
+
       if (mutations.size() > 0
           && mutationsSize + size >= DatastoreV1.DATASTORE_BATCH_UPDATE_BYTES_LIMIT) {
         flushBatch();
```

Comment:
we also need to add the batch size here. In fact, maybe update the batch size within `flushBatch`?